sealtk/core/TimeMap.hpp

```diff
@@ -5,19 +5,33 @@
 #ifndef sealtk_core_TimeMap_hpp
 #define sealtk_core_TimeMap_hpp
 
+#include <sealtk/core/Export.h>
+
 #include <vital/types/timestamp.h>
 
 #include <qtEnumerate.h>
+#include <qtGlobal.h>
 
 #include <QMap>
+#include <QMetaObject>
+#include <QMetaType>
 #include <QSet>
 
+#if 0
+// This is needed to "trick" CMake AUTOMOC into processing this file, since it
+// has meta-object stuff in it, but does not otherwise contain any of the
+// keywords that trigger AUTOMOC.
+Q_GADGET
+#endif
```

Comment:
Please use [`CMAKE_AUTOMOC_MACRO_NAMES`](https://cmake.org/cmake/help/latest/variable/CMAKE_AUTOMOC_MACRO_NAMES.html) instead.

Reply:
Or [`AUTOMOC_MACRO_NAMES`](https://cmake.org/cmake/help/latest/prop_tgt/AUTOMOC_MACRO_NAMES.html).

Reply:
Hmm, that could be better documented; I didn't know it existed, and it isn't linked from [`AUTOMOC`](https://cmake.org/cmake/help/v3.9/prop_tgt/AUTOMOC.html#prop_tgt:AUTOMOC) (which it clearly should be, as [`AUTOMOC_MACRO_NAMES`](https://cmake.org/cmake/help/latest/prop_tgt/AUTOMOC_MACRO_NAMES.html) implies that the former page isn't telling the whole story).

Anyway, see https://github.com/Kitware/qtextensions/pull/28; thanks for the tip!

Reply:
`AUTOMOC_MACRO_NAMES` has been mentioned in the `AUTOMOC` documentation since [3.10](https://cmake.org/cmake/help/v3.10/prop_tgt/AUTOMOC.html). (SEAL-TK currently requires CMake 3.11.)

Reply:
Ah. (Stupid Google indexing old versions :cry:.)